### PR TITLE
[Core]create internal module for thridparty system compatible building

### LINKED ISSUE #44065: [Important Module Removed] internal.cc/internal.h had been remove, but it's related to mobius.
### What happened + What you expected to happen

I'm just working in building new mobius version(with ray lastest 2.9.3) https://github.com/ray-project/mobius/pull/60 . But found internal.cc/internal.h and exported_internal has been removed in this PR https://github.com/ray-project/ray/pull/34697.

I want to know why these files need to be deleted?
Actually thridparty system(based ray core libraries) would better to integrate core worker functions without innner symbols races. So in that time, we create this module for making sure other module can accessing rpc etc. interface by actor handler.

### Versions / Dependencies

>2.5

### Reproduction script

No need script

### Issue Severity

None

### PR DESCRIPTION
**Pull request by ashione**
Exporting internal module

## Related issue number
Closes #44065

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/internal/internal.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    RayFunction &function,
+    int return_num,
+    int max_retries = -1,
+    bool retry_exceptions = false,
+    std::string serialized_retry_exception_allowlist = "");
+
+const stats::TagKeyType TagRegister(const std::string tag_name);
+
+/// Get current actor id via internal.
+const ActorID &GetCurrentActorID();
```
Comment (jjyao) [inline src/ray/internal/internal.h:48]: What are these functions?

**Comment by larrylian** [comment]
@jjyao 
internal.cc/internal.h and exported_internal has been removed in this PR https://github.com/ray-project/ray/pull/34697 (Because it was thought to be unused. However, in reality, Mobius is using it.）

Now it block build the mobius project update  ray version  (old version -> ray lastest 2.9.3) https://github.com/ray-project/mobius/pull/60 . 
I think we should merge this revert PR first, to support the Mobius upgrade of the Ray version.

**Comment by ashione** [comment]
@robertnishihara @raulchen @SongGuyang Anyone can take over ray core module management? Seems no one can go on developing.

**Comment by SongGuyang** [comment]
Seems that the exported internal flags were deleted accidentally. I will merge this PR if there are no more other comments.

**Review by jjyao — APPROVED** [review]
Can we add these code to `mobius` repo?

We should add some comment saying that the code is used by an external repo otherwise it's guaranteed to be removed in the future by someone without context.

File: src/ray/internal/internal.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "ray/common/buffer.h"
+#include "ray/common/id.h"
+#include "ray/core_worker/common.h"
+#include "ray/stats/metric.h"
+
```
Comment (jjyao) [inline src/ray/internal/internal.h:20]: Can you add comments to the code saying where it's used so people don't accidentally delete it again in the future?

Comment (ashione) [reply src/ray/internal/internal.h:20]: > Can you add comments to the code saying where it's used so people don't accidentally delete it again in the future?

Comments had been added in internal header file.

**Review by scv119 — APPROVED** [review]

**Pull request merged**